### DISCUSSION TITLE
Another way for porting

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#24214] [Ideas]
In rare cases, porting becomes difficult, and harder to maintain, e.g. https://github.com/microsoft/vcpkg/pull/24047
What about let the libs install in a native way, and provide paths for that (somehow like conan)?

**Comment by dg0yt** [comment]
Well, building from source is the job of vcpkg? 
I'm not sure what you mean by "install in a native way".

**Reply by playgithub** [reply]
> Well, building from source is the job of vcpkg? I'm not sure what you mean by "install in a native way".

After `cmake --install ...`, keep the installed files untouched instead of what is done by `vcpkg`.
For `vcpkg` to use the a lib, path info for the lib (other info as well if needed) can be saved somewhere.

**Reply by dg0yt** [reply]
> After `cmake --install ...`, keep the installed files untouched instead of what is done by `vcpkg`.
> For `vcpkg` to use the a lib, path info for the lib (other info as well if needed) can be saved somewhere.

This is what you get in addition to plain `make install`:
- a fairly consistent installation layout across platforms (with debug and release variants for VS users),
- check for file name conflicts,
- relocatable, cacheable binary packages,
- rectified exported config (CMake config, CMake find module wrappers, pkg-config files, some \<port\>-config scripts).
  This is the the valid "path info for the lib (other info as well if needed)" you are asking for, and it wouldn't be easier to provide it in a less formal way. 

This is what makes port maintenance hard:
  - combination of installation layout and platform not supported by upstream.
  - combination of exported config and platform not supported by upstream.
  - lack of exported config support from upstream.
  - assumptions about dependencies made by upstream (including vendored dependencies).
  - lack of control over dependencies in upstream build system (developer convenience over packager requirements).
  - plain bugs from upstream.

Unfortunately, there is no established, sufficient way to validate exported config in vcpkg, apart from random aspects tested by depending ports. That's why we see invalid changes and regressions.

**Reply by playgithub** [reply]
I'll choose `vcpkg`, because of serious contributors.
Thanks
